Newsletters: add 315 (2024-08-09)

harding opened the pull request:
- [x] Lede, releases/RCs, topic entries @harding 
- [x] New topic pages for block withholding (extracted from existing pooled mining topic) and replacement cycling
- [x] PR Review Club summary @stickies-v 
- [x] Merge summaries @Gustavojfe

murchandamus commented:
If it is not too late: https://github.com/bitcoin/bitcoin/pull/29775

harding commented:
@Gustavojfe 

> If it is not too late: https://github.com/bitcoin/bitcoin/pull/29775

I can do that one, since it's late-breaking.

LarryRuane commented on _posts/en/newsletters/2024-08-09-newsletter.md:
```diff
@@ -0,0 +1,342 @@
+    shares, the share would need to be received by the pool and
+    converted into a full block before it could be broadcast.
+
+  Towns concludes by describing two of the motivations for fixing the block
```
```suggestion
  Towns concludes by describing two motivations for fixing the block
```

harding replied:
Leaving as-is.  I think "two of the motivations" expresses that there are other motivations; "two motivations" sounds like there's only two motivations.

LarryRuane replied:
Well, then it would be "the two motivations".

harding replied:
Still leaving as-is, for the reason stated.

LarryRuane commented on _posts/en/newsletters/2024-08-09-newsletter.md:
```diff
@@ -0,0 +1,342 @@
+  reliability of [compact block][topic compact block relay]
+  reconstruction.  Many relaying full nodes have been using [BIP152][]
+  compact block relay since the feature was added to Bitcoin Core 0.13.0
+  in 2016.  This allows two peers which have already shared some
```
```suggestion
  in 2016.  This allows two peers who have already shared some
```
(or maybe "peers that have"?)

LarryRuane commented on _posts/en/newsletters/2024-08-09-newsletter.md:
```diff
@@ -0,0 +1,342 @@
+  seen before.  In that case, the node receiving a compact block usually
+  needs to request those transactions from the sending peer and then
+  wait for the peer to respond.  This slows down block propagation.
+  Until a node has all of the transactions in a block, that block can't
```
```suggestion
  Until a node has all the transactions in a block, that block can't
```

harding replied:
Leaving as-is.  It sounds more correct to me with the preposition.

LarryRuane commented on _posts/en/newsletters/2024-08-09-newsletter.md:
```diff
@@ -0,0 +1,342 @@
+- **Replacement cycle attack against pay-to-anchor:** Peter Todd
+  [posted][todd cycle] to the Bitcoin-Dev mailing list about the
+  pay-to-anchor (P2A) output type that is part of the [ephemeral
+  anchors][topic ephemeral anchors] proposal.  P2A is an transaction
```
```suggestion
  anchors][topic ephemeral anchors] proposal.  P2A is a transaction
```

LarryRuane commented on _posts/en/newsletters/2024-08-09-newsletter.md:
```diff
@@ -0,0 +1,342 @@
+  cpfp] fee bumping---especially in multiparty protocols such as LN.
+  However, CPFP fee bumping in LN is currently vulnerable to a
+  counterparty performing a [replacement cycling attack][topic
+  replacement cycling] where the malicious counterparty perfroms a
```
```suggestion
  replacement cycling] where the malicious counterparty performs a
```

LarryRuane commented on _posts/en/newsletters/2024-08-09-newsletter.md:
```diff
@@ -0,0 +1,342 @@
+  sign bip] for creating [scriptless threshold signatures][topic
+  threshold signature] for Bitcoin's implementation of [schnorr
+  signatures][topic schnorr signatures].  This allows a set of wallets
+  that have already performed a set up procedure to securely create
```
```suggestion
  that have already performed a setup procedure to securely create
```

LarryRuane commented on _topics/en/replacement-cycling.md:
```diff
@@ -0,0 +1,122 @@
+_unilateral exit transaction_ (such as an LN _commitment transaction_)
+that they can publish onchain at any time to terminate the fund sharing
+arrangement.  To allow either of them to fee bump the transaction, it
+contains at least one output each of them is able to spend for a [CPFP
```
```suggestion
contains at least one output each of them can spend for a [CPFP
```

LarryRuane commented on _posts/en/newsletters/2024-08-09-newsletter.md:
```diff
@@ -0,0 +1,342 @@
+  benefit from the attack, so there's no direct incentive for third
+  parties to attack P2A outputs.  The attack can be free in the case
+  where the attacker is planning to broadcast their own transaction at a
+  feerate higher than the honest user's P2A spend and the attacker is
+  able to successfully complete the replacement cycle without their
```
```suggestion
  feerate higher than the honest user's P2A spend and the attacker
  can complete the replacement cycle without their
```

bitschmidty commented on _posts/en/newsletters/2024-08-09-newsletter.md:
```diff
@@ -0,0 +1,391 @@
+  sign bip] for creating [scriptless threshold signatures][topic
+  threshold signature] for Bitcoin's implementation of [schnorr
+  signatures][topic schnorr signatures].  This allows a set of wallets
+  that have already performed a set up procedure to securely create
```
Is a reference to https://bitcoinops.org/en/newsletters/2024/07/19/#distributed-key-generation-protocol-for-frost applicable here?

bitschmidty commented on _posts/en/newsletters/2024-08-09-newsletter.md:
```diff
@@ -0,0 +1,391 @@
+  mailing list to announce the availability of a [proposed BIP][frost
+  sign bip] for creating [scriptless threshold signatures][topic
+  threshold signature] for Bitcoin's implementation of [schnorr
+  signatures][topic schnorr signatures].  This allows a set of wallets
```
perhaps signers instead of wallets in this news item?

bitschmidty commented on _posts/en/newsletters/2024-08-09-newsletter.md:
```diff
@@ -0,0 +1,391 @@
+  option that uses testnet3 remains available but is expected to be deprecated
+  and removed in subsequent releases.
+
+- [Core Lightning #7476][] catches up to the latest [BOLT12
```
Perhaps proposed since the BOLT12 PR is still open
```suggestion
- [Core Lightning #7476][] catches up to the latest proposed [BOLT12
```

bitschmidty commented on _posts/en/newsletters/2024-08-09-newsletter.md:
```diff
@@ -0,0 +1,391 @@
+  endorsement][topic htlc endorsement], becoming the first LN implementation to
+  do so, to partially mitigate [channel jamming attacks][topic channel jamming
+  attacks] on the network. This PR enables the relaying of incoming endorsement
+  values, introducing a reputation system for LN where relaying nodes can choose
```
Im not sure this suggestion is true, but I think it is and would be an important distinction

```suggestion
  values, introducing a local reputation system for LN where relaying nodes can choose
```

harding replied:
Had to rewrite a bit to get the local in there in a way that made sense to me, but I agree that it was an important distinction.

bitschmidty commented on _posts/en/newsletters/2024-08-09-newsletter.md:
```diff
@@ -0,0 +1,391 @@
+  preventing network attacks using only CPU mining. All previous soft forks are
+  enabled from the genesis block in testnet4, and the port used is `48333` by
+  default. See Newsletters [#306][news306 testnet4] and [#311][news311 testnet4]
+  for more details on how testnet4 fixes the issues that led to the fall of
```
```suggestion
  for more details on how testnet4 fixes the issues that led to problematic behavior with
```

harding replied:
The Fall of Testnet3 according to ChatGPT  :-)

![A_dramatic_and_heroic_battle_scene_depicting_'The_](https://github.com/user-attachments/assets/2faffc19-303a-4fbd-8f41-2861e543c051)

bitschmidty commented on _topics/en/replacement-cycling.md:
```diff
@@ -0,0 +1,122 @@
+fee bump][topic cpfp].
+
+If Bob wants to terminate the fund sharing, he can broadcast his
+unilateral exit transaction along with a CPFP fee bump.  Mallory has the
```
Maybe?
```suggestion
unilateral exit transaction along with a CPFP fee bumping transaction.  Mallory has the
```

bitschmidty commented on _topics/en/replacement-cycling.md:
```diff
@@ -0,0 +1,122 @@
+sensitive transactions.  For example, forwarded [HTLCs][topic htlc] in
+LN must be resolved within a certain number of blocks.  If Mallory is
+able to use replacement cycling to prevent Bob from resolving HTLCs
+forward to Mallory within a certain amount of time, Mallory can steal
```
```suggestion
forwarded to Mallory within a certain amount of time, Mallory can steal
```

harding commented:
Made edits (or left replies) for all feedback to date, thanks @LarryRuane and @bitschmidty!  Added lede and topic entries, and updated releases/RCs (thanks @notmandatory for the hints about BDK!).  Reviewed and made some edits to @Gustavojfe section, which was particularly great this week (thanks!).

I'll check again later for any additional feedback and to review the PR Review Club section.

stickies-v commented:
Sorry for the review club delay, we only had the meeting last night so couldn't pre-prep notes this time. Added my commit. If there's no time left to review it I'd be okay with dropping it and including it in next week's newsletter instead, if that's better?

vostrnad commented on _posts/en/newsletters/2024-08-09-newsletter.md:
```diff
@@ -0,0 +1,476 @@
+  [# Add PayToAnchor(P2A), OP_1 <0x4e73> as standard output script for
+  spending][review club 30352] is a PR by [instagibbs][gh instagibbs]
+  that introduces a new `TxoutType::ANCHOR` output script type. Anchor
+  outputs have a `OP_1 <0x4e73>` witness program (resulting in a
```
Witness program refers to just the data push part (`0x4e73`):
```suggestion
  outputs have a `OP_1 <0x4e73>` output script (resulting in a
```

stickies-v replied:
Ah right, thanks!

vostrnad commented on _posts/en/newsletters/2024-08-09-newsletter.md:
```diff
@@ -0,0 +1,476 @@
+  be _created_ in a standard transaction? Is that the same as the script
+  types that can be _spent_ in a standard transaction?"
+  a2="All defined `TxoutType`'s except `TxoutType::NONSTANDARD` can be
+  created. All defined `TxoutType`'s except `TxoutType::NONSTANDARD` and
```
(technically)
```suggestion
  created. All defined `TxoutType`'s except `TxoutType::NULL_DATA`, `TxoutType::NONSTANDARD` and
```

harding replied:
Nice catch!

LarryRuane commented on _posts/en/newsletters/2024-08-09-newsletter.md:
```diff
@@ -0,0 +1,476 @@
+  the attacker over the internet.  Exfiltration is mainly considered to
+  be a risk for hardware signing devices that don't have direct access
+  to the internet.  A device that has had its logic corrupted, either
+  through its firmware or through hardware logic, can now quickly exfiltrate a
```
```suggestion
  through its firmware or hardware logic, can now quickly exfiltrate a
```

LarryRuane commented on _posts/en/newsletters/2024-08-09-newsletter.md:
```diff
@@ -0,0 +1,476 @@
+  Optech recommends that anyone using hardware signing devices to protect
+  substantial amounts of money guard against corrupted
+  hardware or firmware, either through the use of exfiltration-resistant
+  signing or through the use of multiple independent devices (e.g. with
```
```suggestion
  signing or multiple independent devices (e.g. with
```

harding replied:
Leaving as-is.  This puts the verb ("use") too far from the second term ("multiple independent devices").

LarryRuane commented on _posts/en/newsletters/2024-08-09-newsletter.md:
```diff
@@ -0,0 +1,476 @@
+
+  Pool miners are paid for submitting units of work, called _shares_,
+  each of which is a _candidate block_ that contains a certain amount of
+  proof of work (PoW).  The expectation is that a portion of those
```
```suggestion
  proof of work (PoW).  The expectation is that some of those
```

LarryRuane commented on _posts/en/newsletters/2024-08-09-newsletter.md:
```diff
@@ -0,0 +1,476 @@
+  attempt mining transactions that the pool doesn't have.  This can make
+  it expensive for the pool to validate miner shares: each share can
+  contain up to several megabytes of transactions that the pool has
+  never seen before, with potentially all of them being transactions
```
```suggestion
  never seen before, all of which may be
```

harding commented:
Reviewed @stickies-v section (looks great, thanks!  No worries about the scheduling) and made edits (or left replies) for all additional feedback (thanks @vostrnad and @LarryRuane !).  I also checked for any additional releases/RCs (none found).

Thanks everyone!

vostrnad commented on _posts/en/newsletters/2024-08-09-newsletter.md:
```diff
@@ -0,0 +1,477 @@
+  what `TxoutType` would a `scriptPubKey` `OP_1 <0x4e73>` be classified
+  as?"
+  a0="Because it consists of a 1-byte push opcode (`OP_1`) and a 2-byte
+  data push (`0x4e73`), it is a valid v1 witness program. Because it is
```
Missed this in my previous review, same reason:
```suggestion
  data push (`0x4e73`), it is a valid v1 witness output. Because it is
```

harding replied:
Fixed, thanks!

bitschmidty reviewed (approved):
ACK lede, topics.
Squashed

Thank you @harding @Gustavojfe @stickies-v for authoring this week and @LarryRuane @vostrnad for the reviews!
